src/glados/core/engine.py

```
@@ -0,0 +1,383 @@
+            pydantic.ValidationError: If the configuration is invalid
+        """
+        path = Path(path)
+
+        # Try different encodings
+        for encoding in ["utf-8", "utf-8-sig"]:
+            try:
+                data = yaml.safe_load(path.read_text(encoding=encoding))
+                break
+            except UnicodeDecodeError:
+                if encoding == "utf-8-sig":
+                    raise ValueError(f"Could not decode YAML file {path} with any supported encoding")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)